### PR TITLE
issue 846: Add --output plain flag for stdout output without JLine

### LINKED ISSUE #846: Add --output plain flag for stdout output without JLine
## Description

Add a new `--output plain` flag to the Wanaku CLI that writes output directly to stdout instead of /dev/tty using JLine. This is essential for testing the Wanaku CLI when running it as a subprocess.

## Motivation

Currently, the CLI uses JLine which writes to /dev/tty, making it difficult to capture output when running Wanaku as a subprocess in automated tests or scripts. The `--output plain` flag would provide a way to bypass JLine and write directly to stdout, enabling proper output capture and testing.

## Technical Details

The root cause is that `WanakuPrinter` (extends JLine's `DefaultPrinter`) calls `terminalInstance()` with `.system(true)`, which opens `/dev/tty` directly, bypassing the subprocess stdout pipe.

**Note:** On CI environments without a TTY, JLine falls back to `DumbTerminal` and writes to stdout correctly. The problem only manifests when a controlling terminal is present (local development, interactive machines).

**Attempted workarounds:** JVM-level flags (`-Dorg.jline.terminal.provider=dumb`, `-Dorg.jline.terminal.dumb=true`) did not resolve the issue when running as a subprocess.

## Expected Behavior

When running:
```bash
wanaku --output plain <command>
```

The CLI should:
- Write all output directly to stdout (not /dev/tty)
- Disable JLine terminal handling
- Produce plain text output suitable for capturing in subprocess calls
- Allow automated testing and scripting of the Wanaku CLI

## Additional Context

This is particularly important for:
- Integration tests that run the CLI in subprocesses
- CI/CD pipelines that need to capture and verify CLI output
- Scripting and automation workflows

### PR DESCRIPTION
**Pull request by luigidemasi**
fix #846

## Summary by Sourcery

Add a configurable plain output mode that routes CLI terminal output through stdout for easier capture by parent processes, and ensure ANSI rendering respects the active terminal.

New Features:
- Introduce a global plain output mode flag in the CLI printer to switch between interactive terminal and stdout-based output.
- Add a --plain command-line option on the base CLI command to enable capturable plain output across commands.

Bug Fixes:
- Ensure exception handling respects the --plain flag by detecting it across subcommand chains and configuring the printer accordingly.

Enhancements:
- Adjust terminal creation to support both interactive and plain modes while maintaining robust fallbacks for non-interactive environments.
- Update markdown and styled message rendering to use the active terminal for ANSI conversion and to omit styling when plain mode is enabled.

### CODE REVIEW

File: cli/src/main/java/ai/wanaku/cli/main/support/WanakuPrinter.java [hunk 1]
```diff
@@ -62,6 +62,23 @@
  */
 public class WanakuPrinter extends DefaultPrinter {
 
+    /**
+     * When {@code true}, the terminal is created with {@code system(false)} so that output
+     * goes through {@code System.out} instead of {@code /dev/tty}.  This keeps ANSI
+     * colors/formatting intact but allows a parent process to capture the output.
+     */
+    private static volatile boolean plainMode = false;
```
[the remark this answers is not in the document]
Comment (luigidemasi) [reply cli/src/main/java/ai/wanaku/cli/main/support/WanakuPrinter.java:70]: Each CLI invocation is a fresh JVM, so it won't happens in production.
It can happens edge cases like Quarkus dev mode or future test scenarios, where multiple commands could run in the same JVM. 
Fixed: reset plainMode to false after every invocation.
